### ISSUE TITLE
Java exception in perl plugin with Idea 2020.2, file lists permanently "loading..." and "Analyzing..." stays forever

### ISSUE THREAD

**Issue by aldem** [Camelcade/Perl5-IDEA#2268]
So, upgrade was not so smooth...

- IntelliJ IDEA 2020.2 (Ultimate Edition), Build #IU-202.6397.94, built on July 27, 2020, Runtime version: 11.0.7+10-b944.20 amd64
- Perl plugin 2020.2
- Windows 10
- Perl 5.26.1 in WSL1 (Ubuntu 18.04)

All projects that were perfectly OK in previous release are now stuck and show no files under "Projects files", it shows now "loading..." and that's all, invalidating caches and restarts works once but any new project shows same problems. Additionally, a message hint (top right corner of the edit window) shows "Analyzing..." - also permanently.

Disabling plugin, or even switching "Perl5" analysis mode to "None" solves the issue, but anything else ("Syntax" or "All problems") produces problems - thus I guess that this is related to plugin itself.

And at least once (I could not reproduce it but I remember that it appeared after enabling plugin) I have got the exception (never had any before):
```
java.lang.NullPointerException
	at com.perl5.lang.perl.idea.modules.PerlSourceRootType.getRoots(PerlSourceRootType.java:52)
	at com.perl5.lang.perl.idea.modules.PerlSourceRootType.getRoots(PerlSourceRootType.java:46)
	at com.perl5.lang.perl.idea.project.PerlProjectManager.lambda$new$0(PerlProjectManager.java:83)
	at com.intellij.util.containers.FactoryMap$1.create(FactoryMap.java:189)
	at com.intellij.util.containers.FactoryMap.get(FactoryMap.java:67)
	at com.perl5.lang.perl.idea.project.PerlProjectManager.getModulesRootsOfType(PerlProjectManager.java:212)
	at com.perl5.lang.perl.idea.project.PerlProjectManager.getModulesLibraryRoots(PerlProjectManager.java:208)
	at com.perl5.lang.perl.idea.project.PerlProjectManager.lambda$resetProjectCaches$7(PerlProjectManager.java:187)
	at com.intellij.openapi.util.AtomicNotNullLazyValue$1.compute(AtomicNotNullLazyValue.java:59)
	at com.intellij.openapi.util.AtomicNotNullLazyValue.getValue(AtomicNotNullLazyValue.java:37)
	at com.perl5.lang.perl.idea.project.PerlProjectManager.getProjectLibraries(PerlProjectManager.java:223)
	at 
com.perl5.lang.perl.idea.configuration.settings.sdk.PerlLibraryProvider.getAdditionalProjectLibraries(PerlLibraryProvider.java:36)
...rest is skipped as it mostly points to Idea SDK mostly...
```

And another one was lurking in status tray, when I was checking on absolutely empty project with single empty .pl file (so the problems are not in my projects either):
```
java.lang.Throwable: This supposed to be invoked from pooled thread
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:143)
	at com.perl5.lang.perl.idea.sdk.host.PerlHostCacheCleaner.runActivity(PerlHostCacheCleaner.java:56)
	at com.intellij.ide.startup.impl.StartupManagerImpl.runActivity(StartupManagerImpl.java:293)
	at com.intellij.ide.startup.impl.StartupManagerImpl.access$000(StartupManagerImpl.java:53)
	at com.intellij.ide.startup.impl.StartupManagerImpl$1.extensionAdded(StartupManagerImpl.java:265)
	at com.intellij.ide.startup.impl.StartupManagerImpl$1.extensionAdded(StartupManagerImpl.java:260)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.notifyListeners(ExtensionPointImpl.java:765)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.lambda$registerExtensions$14(ExtensionPointImpl.java:983)
	at com.intellij.ide.plugins.DynamicPlugins.loadPluginDescriptor(DynamicPlugins.kt:775)
	at com.intellij.ide.plugins.DynamicPlugins.access$loadPluginDescriptor(DynamicPlugins.kt:105)
	at com.intellij.ide.plugins.DynamicPlugins$loadPlugin$1.run(DynamicPlugins.kt:694)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteAction(ApplicationImpl.java:980)
	at com.intellij.ide.plugins.DynamicPlugins.loadPlugin(DynamicPlugins.kt:690)
	at com.intellij.ide.plugins.DynamicPlugins.loadPlugin$default(DynamicPlugins.kt:678)
	at com.intellij.ide.plugins.DynamicPlugins.loadPlugin(DynamicPlugins.kt)
	at com.intellij.ide.plugins.PluginEnabler.updatePluginEnabledState(PluginEnabler.java:69)
	at com.intellij.ide.plugins.newui.MyPluginModel.applyEnableDisablePlugins(MyPluginModel.java:242)
	at com.intellij.ide.plugins.newui.MyPluginModel.apply(MyPluginModel.java:194)
	at com.intellij.ide.plugins.PluginManagerConfigurable.apply(PluginManagerConfigurable.java:1615)
	at com.intellij.openapi.options.ex.ConfigurableWrapper.apply(ConfigurableWrapper.java:182)
...
```

"Analyzing.." seems to stick forever wherever I have any issue with my code, and disappears when there are no obvious issues (like syntax errors), though _sometimes_ it disappears after 20-30 seconds and indeed shows problems found.

However, on a positive note - symbol lookups are much quicker (and correct), though with analysis switched off it is not so useful :)

**Comment by hurricup** [comment]
The last one is a IDEA problem, should be fixed in bugfix update. 
Will investigate the first one, thank you.

**Comment by hurricup** [comment]
I guess I need a full clean log. 
Please:
1. enable plugin
2. Choose invalidate caches (without restart)
3. Shutdown IDE
4. Go to logs folder
5. Remove everything in there
6. Start IDE
7. Open project in question
8. If problem persists. Wait for a min and then shutdown IDE
9. Pack logs directory with fresh logs and attach here or send me to my personal email `hurricup` on gmail.

**Comment by aldem** [comment]
I should have looked into logs before... The first exception there was caused by Haxe plugin - which had issues in 2020.1 and thus was disabled automatically, but it didn't allow me to remove nor disable it (it was simply not shown), but now I was able to disable it - and it _seems_ that Perl plugin works without issues (though there were no haxe-related files in any of my projects).

I will check with other (bigger and multilanguage) projects a bit later and will let you know if there are any issues, as for now it looks stable.

Sorry for the buzz, next time I will check logs first and try to eliminate possible offenders before reporting  (I didn't expect that plugins may damage something beyond themselves or project files).
